contracts/contracts/lib/access/access_control.tolk

```diff
@@ -411,9 +429,24 @@ fun AccessControl<T>.grantRole(mutate self, queryId: uint64, sender: address, ro
 /// - the `sender` must have `role`'s admin role.
 ///
 /// Emits a {AccessControl_RoleRevoked} event if `role` was revoked.
-fun AccessControl<T>.revokeRole(mutate self, queryId: uint64, sender: address, role: int, account: address): bool {
+fun AccessControl<T>.revokeRole(
+    mutate self,
+    queryId: uint64,
+    sender: address,
+    role: int,
+    account: address,
+    returnExcess: bool
+): bool {
     self.requireRole(self.getRoleAdmin(role), sender);
-    return self._revokeRole(queryId, sender, role, account);
+    val success = self._revokeRole(queryId, sender, role, account);
+
+    if (!success) {
+        return false; // nothing to do, role was not updated
+    }
+
+    val body = AccessControl_RoleRevoked {queryId, role, account, sender}.toCell();
```

Comment:
I'll move this to the underlying (no access requirements) functions, so when users skip the checks they still emit messages on state updates.